src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,2089 @@
+
+            elif j == 1:  # X basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.X))
+                expectation_val = GST_measure_1qb(new_circ, k, j)
+
+            elif j == 2:  # Y basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.Y))
+                expectation_val = GST_measure_1qb(new_circ, k, j)
+
+            elif j == 3:  # Z basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.Z))
+                expectation_val = GST_measure_1qb(new_circ, k, j)
+
+            matrix_jk_1qb[j, k] = expectation_val
+
+    matrix_jk_1qb = np.matrix(matrix_jk_1qb)
+
+    return matrix_jk_1qb
+
+
+def GST_2qb(
+    NshotsGST=1e4,
+    noise_model=None,
+    save_data=None,
+    class_qubit_gate=None,
+    ctrl_qb=None,
+    targ_qb=None,
+    theta=None,
+    operator_name=None,
+    backend=None,
+):
+    """Runs gate set tomography for a two qubit gate/circuit.
+
+        Args:
+        NshotsGST (int, optional): Number of shots used in Gate Set Tomography.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied
+            to simulate noisy computation.
+        save_data: To save gate set tomography data. If None, skip.
+        class_qubit_gate: Identifier for the type of gate given in qibo.raw. If None,
+            do gate set tomography with empty circuit.
+        ctrl_qb: Parameter for the control qubit.
+        targ_qb: Parameter for the target qubit.
+        theta: Identifier for the gate parameter given in qibo.raw. If None, do gate
+            set tomography with only gate provided by class_qubit_gate.
+        operator_name: Name of gate given in qibo.raw used for saving data.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+
+    Returns:
+        numpy.matrix: Matrix with elements jk equivalent to either Tr(Q_j rho_k) or Tr(Q_j O_l rho_k).
+    """
+
+    import numpy as np
+
+    from qibo import gates
+    from qibo.backends import GlobalBackend
+    from qibo.models import Circuit
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    if class_qubit_gate is None:
+        print("Empty circuit without any operator (2 qb)")
+
+    def GST_measure_2qb(circuit, k, j):
```

Comment:
Also this one, I would pull outside

Reply:
Yup! I've done the same as the ```GST_measure_1qb```. It's now outside of the ```GST_2qb``` function.